applicationinsights/channel/AsynchronousSender.py

```diff
@@ -123,6 +129,14 @@ def _run(self):
                 # if we didn't get any items from the queue, we're done here
                 if len(data) == 0:
                     break
+                
+                # Drop extra events from the queue until it no larger than send_buffer_size.
+                # This is a necessary evil to prevent the backlog from bloating forever in bandwidth-limited environments
+                eventsToDrop = max(0, local_queue._queue.qsize() - self._send_buffer_size)
```

Comment:
Is this to enforce the queue doesn't get above 500 items? The drop logic looks based off of the current number of items in the queue rather than the maximum limit. If the max queue limit is specified when the queue is constructed, python should restrict additional adds to the queue regardless. If this is to drop the items we put in the queue that got sent this cycle, the .get() calls a few lines above should be doing that already.